manifests/dcgm/telegraf.pp

```diff
@@ -22,106 +14,39 @@
     $ensure_parm = 'absent'
   }
 
+  file { '/etc/telegraf/telegraf.d/dcgmd.conf':
+    ensure  => 'file',
```

Comment:
You'll want to change this line to read:
ensure => $ensure_parm,

That way if a node switches from enabled to disabled it will cleanup the file and also restart telegraf

Reply:
Done, I've made change on my local machine.  Will wait to commit once I've gathered any other input then will rebase/squash commits all in one shot